File: 프로그래머스/unrated/181918. 배열 만들기 4/배열 만들기 4.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        LinkedList<Integer> stk = new LinkedList<Integer>();
        int i=0;
        while(i< arr.length){
            if(stk.size() == 0){
                stk.add(arr[i]);
                i++;
            }
            else{
                if(stk.get(stk.size()-1) >= arr[i]){
                    stk.removeLast();
                } else{
                    stk.add(arr[i]);
                    i++;
                }
            }
        }
        return stk.stream().mapToInt(t->t).toArray();
    }
}

File: 프로그래머스/unrated/181918. 배열 만들기 4/배열 만들기 4.py
from collections import deque

def solution(arr):
    stk = deque([])
    i = 0
    while i < len(arr):
        if len(stk) == 0:
            stk.append(arr[i])
            i+=1
        else:
            if stk[-1] < arr[i]:
                stk.append(arr[i])
                i+=1
            else:
                stk.pop()
    return list(stk)